core-bundle/src/Twig/Runtime/ContentUrlRuntime.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Twig\Runtime;
+
+use Contao\CoreBundle\Routing\ContentUrlGenerator;
+use Symfony\Bridge\Twig\Extension\RoutingExtension;
+use Symfony\Component\Routing\Exception\ExceptionInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Twig\Extension\RuntimeExtensionInterface;
+use Twig\Node\Node;
+
+final class ContentUrlRuntime implements RuntimeExtensionInterface
+{
+    /**
+     * @internal
+     */
+    public function __construct(
+        private readonly ContentUrlGenerator $urlGenerator,
+        private readonly RoutingExtension $routingExtension,
+    ) {
+    }
+
+    public function generate(object $content, array $parameters = [], bool $relative = false): string|null
+    {
+        try {
+            return $this->urlGenerator->generate($content, $parameters, $relative ? UrlGeneratorInterface::ABSOLUTE_PATH : UrlGeneratorInterface::ABSOLUTE_URL);
+        } catch (ExceptionInterface) {
+            return null;
+        }
+    }
+
+    public function isUrlGenerationSafe(Node $argsNode): array
```

Comment:
This does not work:

<img width="893" alt="" src="https://github.com/contao/contao/assets/1192057/91077024-71d1-49ae-8143-4b4d48370f35">

Symfony puts the `isUrlGenerationSafe()` methods directly into the extension classes (e.g. `RoutingExtension`). Can we do the same?

Reply:
I don't really wanna copy and paste the whole code. And I don't really like adding _yet another_ dependency to the `ContaoExtension`. That's why I added a runtime in the first place. But maybe @m-vo or @ausi know what would be better.

Reply:
The only reason we are injecting `Symfony\Bridge\Twig\Extension\RoutingExtension` in the first place is to not have to copy its `isUrlGenerationSafe()` method - which actually has no further dependencies and could be called statically. Injecting the `RoutingExtension` with its own full dependencies just to re-use a method is may be overkill. May be either we copy the method after all - or we use reflection to call the method statically?

Reply:
IMHO it is correct to copy the code because this is a security feature and we cannot really guarantee that changes to their method still have the correct results for us. 

We also might want/nerd to change the return array to `['html', 'contao_html']`...